src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -141,6 +164,143 @@ public void traverse() {
+                                                 cramRefFile);
+    }
+
+    @DataProvider
+    public Object[][] provideForGetMasterSequenceTest() {
+
+        final String dictFileName = ReferenceUtils.getFastaDictionaryFileName(v37_chr17_1Mb_Reference);
+        final SAMSequenceDictionary dict = ReferenceUtils.loadFastaDictionary(new File(dictFileName));
+
+        return new Object[][] {
+                { null, NA12878_chr17_1k_BAM, null },
+                { dictFileName, NA12878_chr17_1k_BAM, dict },
+        };
+    }
+
+    @Test(dataProvider = "provideForGetMasterSequenceTest")
+    public void testGetMasterSequence(String masterSequenceFileName, String inputFileName, SAMSequenceDictionary expectedDict) {
+        final GATKTool tool = new TestGATKToolWithSequenceDictionary();
+        final CommandLineParser clp = new CommandLineArgumentParser(tool);
+
+        final String[] args = (masterSequenceFileName == null)
+                ? new String[] { "--input", inputFileName, }
+                : new String[] { "--sequenceDictionary", masterSequenceFileName, "--input", inputFileName, };
+
+        clp.parseArguments(System.out, args);
+
+        // This initializes our tool
+        tool.onStartup();
+
+        // Make sure that the master sequence dictionary dictionary is the expected dictionary
+        Assert.assertEquals( tool.getMasterSequenceDictionary(), expectedDict );
+    }
+
+    @DataProvider
+    public Object[][] provideMultipleSequenceDictionaries() {
+
+        final String v37_chr19DictFileName = ReferenceUtils.getFastaDictionaryFileName(v37_chr17_1Mb_Reference);
+        final SAMSequenceDictionary v37_chr19Dict = ReferenceUtils.loadFastaDictionary(new File(v37_chr19DictFileName));
+
+        final String b37_20_21DictFile = ReferenceUtils.getFastaDictionaryFileName(b37_reference_20_21);
+        final SAMSequenceDictionary b37_20_21Dict = ReferenceUtils.loadFastaDictionary(new File(b37_20_21DictFile));
+
+        final String hg19MiniDictFileName = ReferenceUtils.getFastaDictionaryFileName(hg19MiniReference);
+        final SAMSequenceDictionary hg19MiniDict = ReferenceUtils.loadFastaDictionary(new File(hg19MiniDictFileName));
+
+        return new Object[][] {
+                { v37_chr19DictFileName, "--input", NA12878_chr17_1k_BAM, v37_chr19Dict, null, null },
+                { b37_20_21DictFile, "--input", BQSR_WGS_B37_CH20_21_10M_100_CRAM, b37_20_21Dict, "--reference", b37_reference_20_21 },
+                { hg19MiniDictFileName, "--reference", hg19MicroReference, hg19MiniDict, null, null },
+        };
+    }
+
+    @Test(dataProvider = "provideMultipleSequenceDictionaries")
+    public void testGetBestAvailableSequenceDictionaryWithMasterDictionary(String sequenceFileName,
+                                                               String otherSeqArg, String otherSequenceFile,
+                                                               SAMSequenceDictionary expectedDict,
```

Comment:
Maybe refactor this to put the `expectedDict` as the final argument -- putting it in the middle like this makes the `DataProvider` test cases harder to read/understand.